ascend-sdk/src/main/java/ai/evolv/ascend/android/Allocator.java

```diff
@@ -2,26 +2,62 @@
 
 import android.support.annotation.NonNull;
 
-import ai.evolv.ascend.android.exceptions.AscendAllocationException;
-import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
```

Comment:
Avoid third-party dependencies if possible.

Reply:
Im going to keep the guava futures for now, ill see what I can do about removing Guava from the project.